### DISCUSSION TITLE
Open remote files from link

### DISCUSSION THREAD

**Discussion by MesaBul** [sumatrapdfreader/sumatrapdf#3651] [Q&A]
Hi all.

I have a pdf with link to remote files that I have mounted on a drive.

However when I click them SumatraPDF tells me that it cannot find the file because it is looking for it from C:/path.

There is no way to remove from that C: and put another drive. The file always tries to open from the folder I'm in and adds the path behind it again.

How can I do it?

Thank you very much.

**Comment by GitHubRulesOK** [comment]
PDF is basically unchangeable "Fixed" content as designed for universal "printOUT"
 so files interrelationships should be "relative" to that
 i.e a sibling file will primarily either be in same folder or at a web (URL) location (http[s]://)
The basic mechanisms in a PDF are thus
- does the text match a valid external URL (http / https / email)
- OR a relative PAGE (goto 404)
- or goto a sister page via filename `goto companion.pdf page 404` 

There are some variations on those but without your sample it is hard to say why you think SumatraPDF is not using one of those methods.

**Reply by MesaBul** [reply]
Hi!!

Thanks for your fast answers! And sorry for the delay.

They are files created from a Word file and have buttons with hyperlinks to other pdfs or videos on a server.

The file path is correct, for example with Microsoft Edge I can open it, but from SumatraPDF it tries to open the file from "SumatraPDF path"/"hyperlink path".

I have tried /../ but I can't get beyond C:.

Greets from Barcelona.

**Comment by kjk** [comment]
Is it a PDF or some other file?

Not opening absolute paths is for security: we don't want an attacker to create a PDF file to e.g. an .exe or some other malicious file and trick the user to launch it.

But maybe I could restrict that for files that would open in SumatraPDF, like PDF files etc.

Relevant code: `LinkHandler::LaunchFile`

**Reply by MesaBul** [reply]
Hi!

Thanks for your fast answers! And sorry for the delay.

I've installed a computer running Windows and my goal is to show to my colleagues a pdf. This pdf teach how to perform some actions in our lab. In this pdf there are going to be links to other files of our LAN (only pdf's, powerpoints, pictures and videos).

Sumatra is the default app for opening pdf's and pictures, so far so good. The Powerpoints are going to be always in ppsx format and the videos are going to be opened with VideoLAN.

I'm fine with security limitations, but the pdf is never going to have a link to an exe. Just pdf, jpg, png, mp4, avi, mov, ppsx, pps...

And all the files are going to be stored in my server, there's no need to gain access to the internet.

Is there anything that we can do in sumatrapdfrestrict.ini or in SumatraPDF-settings.txt that we can edit to allow what we are trying to archieve?

We also have tried to edit manually the links to avoid absolute paths and changed all with relative paths, but we haven't been able to make it work neither (it works fine with files in the same directory, but we can not get into a parent folder).

Greets from Barcelona

**Comment by cesta13** [comment]
I have the same problem.
I make list to other pdf files with Excel.
If I open pdf file with link to PDF file on some location on computer or mapped drive with other readers such as Acrobat reader ... or web browser such as vivaldi, firefox ... I can open linked file with same program. 

But SumatraPDF want to open file in path of saved oridinal file (c:\temp\pdf) + path defined in link (\\server\prod\pdf)
Looking for file in c:\temp\pdf\server\prod\pdf

Is there a solution to setup Sumatra to open file in absolut location?

Thanks

**Reply by GitHubRulesOK** [reply]
@cesta13 

SumatraPDF is more secure than other apps so has no active JavaScripting such as Acrobat or chrome or Firefox may use.
Likewise it does not run absolute addresses that can be used by exploits. 
For that reason it is more limited to as described above to absolute URLs
or locally "relative" siblings or descendants files. However are you sure it cannot reach other targets here it is opening a file on another computer using absolute mapped folder behind my back.

![image](https://github.com/user-attachments/assets/0c96a698-526b-4749-9171-4f0039367140)
